src/components/Api.js

```diff
@@ -0,0 +1,220 @@
+export default class Api {
+  constructor(options) {
+    this._url = options.url;
+    this.authorization = options.authorization;
+  }
+
+  getCurrentUser(endPoint, DOMLocations) {
+    return fetch(`${this._url}${endPoint}`, {
+      method: "GET",
+      headers: {
+        authorization: this.authorization,
+      },
+    })
+      .then((res) => {
```

Comment:
Needs correction: The code for processing server response gets duplicated. You should declare a method and reuse it

```js
_processResponse = (res) => { 
      if (res.ok) { 
        return res.json(); 
      } 
      return Promise.reject(`Error: ${res.status}`); 
}

getInitialCards() { 
  ...
  .then(this._processResponse)
}
```